geth/api/api.go

```diff
@@ -121,21 +124,46 @@ func (api *StatusAPI) CallRPC(inputJSON string) string {
 // BIP44-compatible keys are generated: CKD#1 is stored as account key, CKD#2 stored as sub-account root
 // Public key of CKD#1 is returned, with CKD#2 securely encoded into account key file (to be used for
 // sub-account derivations)
-func (api *StatusAPI) CreateAccount(password string) (address, pubKey, mnemonic string, err error) {
-	return api.b.AccountManager().CreateAccount(password)
+func (api *StatusAPI) CreateAccount(password string) common.AccountInfo {
```

Comment:
While the idea of bundling the account information into one struct may be nice the error at least has to stay outside:

```
CreateAccount(string) (common.AccountInfo, error)
```

The `StatusAPI` is the pure Go API and has to follow Go idioms.

Reply:
Agreed not Go convention. I wasn't sure which way to go as returning multiple values will mean StatusAPI won't be able to be used by gomobile as mentioned in #440 and #391

Assuming refactoring to return separate err, is leaving the populating of the struct (common.AccountInfo.Error) with the error message acceptable, as this also doesn't feel all that Go like, but simplifies the binding side a lot.

Reply:
You still could fill the `AccountInfo.Error` as you do.